src/main/java/com/shapesecurity/shift/es2016/scope/ScopeAnalyzer.java

```diff
@@ -514,7 +514,7 @@ private State finish(@Nonnull Node astNode, @Nonnull Scope.Type scopeType, boole
                     for (Pair<String, ImmutableList<Declaration>> entry2 : this.blockScopedDeclarations.merge(this.functionDeclarations, ImmutableList::append).entries()) {
                         String name2 = entry2.left();
                         ImmutableList<Declaration> declarations2 = entry2.right();
-                        ImmutableList<Reference> references2 = freeIdentifiers.get(name2).orJust(ImmutableList.empty());
+                        ImmutableList<Reference> references2 = freeIdentifiers.get(name2).map(referenceList -> (ImmutableList<Reference>)referenceList).orJust(ImmutableList.empty());
```

Comment:
I don't like this casting `map` here since it's not a free cast. Does

```js
(ImmutableList<Reference>) freeIdentifiers.get(name2).orJust(ImmutableList.empty())
```

or

```js
freeIdentifiers.get(name2).orJust<ImmutableList<Reference>>(ImmutableList.empty())
```

work?